yarpl/test/FlowableTest.cpp

```diff
@@ -400,5 +403,56 @@ TEST(FlowableTest, FlowableCompleteInTheMiddle) {
   EXPECT_EQ(std::size_t{1}, subscriber->values().size());
 }
 
+class RangeCheckingSubscriber : public Subscriber<int32_t> {
+ public:
+  explicit RangeCheckingSubscriber(int32_t total, folly::fibers::Baton& b)
+      : total_(total), on_complete_(b) {}
+
+  void onSubscribe(Reference<Subscription> subscription) override {
+    Subscriber<int32_t>::onSubscribe(subscription);
+    subscription->request(total_);
+  }
+
+  void onNext(int32_t val) override {
+    EXPECT_EQ(val, current_);
+    current_++;
+  }
+
+  void onComplete() override {
+    Subscriber<int32_t>::onComplete();
+    EXPECT_EQ(total_, current_);
+    on_complete_.post();
+  }
+
+ private:
+  int32_t current_{0};
+  int32_t total_;
+  folly::fibers::Baton& on_complete_;
+};
+
+TEST(FlowableTest, FlowableFromDifferentThreads) {
+  auto flowable = Flowable<int32_t>::create([&](auto& subscriber, int64_t req) {
+    EXPECT_GE(req, 10000);
+    auto t1 = std::thread([&] {
```

Comment:
nit: I believe we use ScopedEventBaseThread in our examples. Would you mind updating the test to use that also?